src/common/checksum.c

```diff
@@ -486,3 +502,39 @@ checksum_sgl_cb(uint8_t *buf, size_t len, void *args)
 
 	return daos_csummer_update(obj, buf, len);
 }
+
+int
+daos_csum_check_sgl(const daos_iod_t *iod, d_sg_list_t *sgl)
+{
+	struct daos_csummer	*csummer;
+	daos_csum_buf_t		*csum_bufs;
+	daos_csum_buf_t		*csum = iod->iod_csums;
+	int			 rc;
+
+	if (!dcb_is_valid(csum))
+		return 0;
+
+	rc = daos_csummer_type_init(&csummer,
+				    (enum DAOS_CSUM_TYPE) csum->cs_type,
+				    csum->cs_chunksize);
+	if (rc != 0) {
+		D_ERROR("Issue initializing csummer. "
+			"Unable to check data. Returning success so checksum "
+			"doesn't kill IO.");
+
+		return 0;
```

Comment:
This seems like a pretty bad problem?   I'm not sure what the message means here.  Don't we want to return some error to the user if we were unable to validate the checksum?

On another note, what are the valid cases where the csummer would fail to initialize?  Can we move the failure off of the critical I/O path?

Reply:
Ya, I really want to avoid killing the IO for checksum related issues (outside of discovering corruption). I'll look into initializing the csummer elsewhere. I didn't really like initializing it for every update anyway ... I just wasn't sure where else to do it. I'll look a little more :-)